api/apps/dataset_api.py

```diff
@@ -135,53 +137,83 @@ def list_datasets():
 
 @manager.route('/<dataset_id>', methods=['DELETE'])
 @login_required
-@validate_request("dataset_id")
 def remove_dataset(dataset_id):
     req = request.json
     try:
-        kbs = KnowledgebaseService.query(
-            created_by=current_user.id, id=req["dataset_id"])
-        if not kbs:
+        datasets = KnowledgebaseService.query(
+            created_by=current_user.id, id=dataset_id)
+        if not datasets:
             return construct_json_result(
                 data=False, message=f'Only owner of knowledgebase authorized for this operation.',
                 code=RetCode.OPERATING_ERROR)
 
-        for doc in DocumentService.query(kb_id=req["dataset_id"]):
-            if not DocumentService.remove_document(doc, kbs[0].tenant_id):
+        for doc in DocumentService.query(kb_id=dataset_id):
+            if not DocumentService.remove_document(doc, datasets[0].tenant_id):
                 return construct_json_result(
```

Comment:
Shouldn't return error, but printout a warning log. The removal of other documents and dataset shouldn't be interrupted.